src/fparser/two/Fortran2008.py

```diff
@@ -108,6 +116,151 @@ class Program_Unit(Program_Unit_2003):  # R202
     subclass_names.append("Submodule")
 
 
+
+class Executable_Construct(Executable_Construct_2003):  # R213
+    # pylint: disable=invalid-name
+    '''
+    Fortran 2003 rule R213
+    executable-construct is action-stmt
+                         or associate-construct
+                         or block-construct
+                         or case-construct
+                         or critical-construct
+                         or do-construct
+                         or forall-construct
+                         or if-construct
+                         or select-type-construct
+                         or where-construct
+
+    Associated constraints are:
+
+    "C201 (R208) An execution-part shall not contain an end-function-stmt,
+          end-mp-subprogram-stmt, end-program-stmt, or end-subroutine-stmt."
+
+    NB: The new block-construct and critical-construct are not yet implemented.
```

Comment:
Please could you add a new Issue and a TODO referencing it here.

Reply:
See #320